### PR TITLE
Include image vulnerability information in ImageSummary

### PR DESCRIPTION
**Pull request by andaaron**
**What type of PR is this?**

cleanup
feature

**Which issue does this PR fix**:
#750 

**What does this PR do / Why do we need it**:

Return this data as part of GlobalSearch and RepoListWithNewestImage query results.
This commit also includes refactoring of the CVE scanning logic in order to better encapsulate trivy specific logic, remove CVE scanning logic from the graphql resolver.

**Does this PR introduce any user-facing change?**:

There's an additional property for several searches.

Signed-off-by: Andrei Aaron <andaaron@cisco.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/extension_search.go [hunk 1]
```diff
@@ -18,7 +18,12 @@ import (
 	"zotregistry.io/zot/pkg/storage"
 )
 
-func EnableSearchExtension(config *config.Config, log log.Logger, rootDir string) {
+// We need this object to be a singleton as read/writes in the CVE BD may
```
Comment (shimish2) [inline pkg/extensions/extension_search.go:21]: BD --> DB ?

File: pkg/extensions/search/cve/cve.go [hunk 2]
```diff
@@ -1,205 +1,202 @@
...
+	imageCVESummary := ImageCVESummary{
+		Count:       0,
+		MaxSeverity: "UNKNOWN",
+	}
 
-						return []ImageInfoByCVE{}, err
-					}
+	isValidImage, err := cveinfo.Scanner.IsImageFormatScannable(image)
+	if !isValidImage {
+		return imageCVESummary, err
```
Comment (shimish2) [inline pkg/extensions/search/cve/cve.go:182]: We should cover this. I think there should be a test case already for that.

File: pkg/extensions/search/cve/trivy/scanner.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+	var tCtx *trivyCtx
+
+	var ok bool
+
+	var rootDir string
+
+	// Get corresponding CVE trivy config, if no sub cve config present that means its default
+	tCtx, ok = scanner.cveController.SubCveConfig[prefixName]
+	if ok {
+		imgStore := scanner.storeController.SubStore[prefixName]
```
Comment (shimish2) [inline pkg/extensions/search/cve/trivy/scanner.go:129]: Please cover this also.

File: pkg/extensions/search/cve/cve.go (hunk 2 above)
Comment (andaaron) [reply pkg/extensions/search/cve/cve.go:182]: Will do. This is actually the new method I added to count the CVEs. Will look at the test for `GetCVEListForImage` which is the older method and has coverage.

**Comment by andaaron** [comment]
Rebased, still need to rerun tests, and make changes according to the review.

**Review by shimish2 — APPROVED** [review]
lgtm

File: pkg/extensions/search/common/common.go [hunk 4]
```diff
@@ -66,18 +66,34 @@ func GetFixedTags(allTags, infectedTags []TagInfo) []TagInfo {
 		return allTags[i].Timestamp.Before(allTags[j].Timestamp)
 	})
 
-	latestInfected := TagInfo{}
+	earliestInfected := infectedTags[0]
```
Comment (rchincha) [inline pkg/extensions/search/common/common.go:69]: s/Infected/Vulnerable/g
or 
s/Infected/Affected/g

File: pkg/extensions/search/cve/cve.go [hunk 5]
```diff
@@ -1,205 +1,207 @@
...
+	GetImageListWithCVEFixed(repo, cveID string) ([]common.TagInfo, error)
+	GetCVEListForImage(image string) (map[string]cvemodel.CVE, error)
+	GetCVESummaryForImage(image string) (ImageCVESummary, error)
+	UpdateDB() error
 }
 
-// UpdateCVEDb ...
-func UpdateCVEDb(dbDir string, log log.Logger) error {
-	return operation.DownloadDB("dev", dbDir, false, false, false)
+type Scanner interface {
```
Comment (rchincha) [inline pkg/extensions/search/cve/cve.go:24]: Good stuff and this becomes important if trivy stops being opensource or drops support

**Review by rchincha — APPROVED** [review]
lgtm

**Review by shimish2 — APPROVED** [review]
lgtm

**Pull request merged**